subprojects/docs/src/docs/userguide/authoring-builds/multi-project/declaring_dependencies_between_subprojects.adoc

```diff
@@ -0,0 +1,134 @@
+│   └── build.gradle.kts
+└── settings.gradle.kts
+----
+=====
+====
+
+We have the projects `shared`, `api` and `person-service`.
+The `person-service` project has a dependency on the other two projects.
+The `api` project has a dependency on the `shared` project.
+It has no build script and gets nothing injected by another build script.
+We use the `:` separator to define a project path.
+Consult the DSL documentation of link:{groovyDslPath}++/org.gradle.api.initialization.Settings.html#org.gradle.api.initialization.Settings:include(java.lang.String[])++[Settings.include(java.lang.String[\])] for more information about defining project paths.
+
+[[javadependencies_2]]
+====
+include::sample[dir="snippets/multiproject/dependencies-java/groovy",files="settings.gradle[];buildSrc/src/main/groovy/myproject.java-conventions.gradle[];api/build.gradle[];shared/build.gradle[];services/person-service/build.gradle[]"]
+include::sample[dir="snippets/multiproject/dependencies-java/kotlin",files="settings.gradle.kts[];buildSrc/src/main/kotlin/myproject.java-conventions.gradle.kts[];api/build.gradle.kts[];shared/build.gradle.kts[];services/person-service/build.gradle.kts[]"]
+====
+
+Shared build logic is extracted into a _convention plugin_ that is applied in the subprojects' build scripts that also define project dependencies.
+A project dependency is a special form of an execution dependency.
+It causes the other project to be built first and adds the jar with the classes of the other project to the classpath.
+It also adds the dependencies of the other project to the classpath.
+You can trigger a `gradle :api:compile`. First the `shared` project is built and then the `api` project is built.
+Project dependencies enable partial multi-project builds.
+
+
+[[sec:depending_on_output_of_another_project]]
+== Depending on output produced by another project
+
+<<#sec:project_jar_dependencies,Project dependencies>> model dependencies between modules.
+Effectively, you are saying that you depend on the main output of another project. In a Java-based project that's usually a JAR file.
+
+Sometimes you may want to depend on an output produced by another task.
+In turn, you'll want to make sure that the task is executed beforehand to produce that very output.
+Declaring a task dependency from one project to another is a poor way to model this kind of relationship and introduces unnecessary coupling.
+The recommended way to model such a dependency is to produce the output, mark it as an "outgoing" artifact or add it to the output of the `main` source set which you can depend on in the consuming project.
+
+Let's say you are working in a multi-project build with the two subprojects `producer` and `consumer`.
+The subproject `producer` defines a task named `buildInfo` that generates a properties file containing build information e.g. the project version.
+You can then map the task provider to its output file and Gradle will automatically establish a task dependency.
```

Comment:
❓ I wonder if we should add a listing showing the "BuildInfo" task here to have a complete example?

If you really wanna understand what is going on, I think you are a bit lost as this is not explained. You might wonder if it is a built-in task of Gradle.